js/scan_worker.js

```diff
@@ -36,19 +36,28 @@ function get_runtime_info() {
 
 function set_prefs(prefs) {
     for (let key in prefs) {
-        let value = prefs[key];
-        switch (typeof(value)) {
+        let prop = prefs[key].split(";")[0];
+        let value = prefs[key].split(";")[1];
+
+    	// Pref values are passed in as strings and must be examined
```

Comment:
There are weird whitespaces in this line and the next, probably mixing spaces and tabs. Switch to all spaces.